pallets/vesting-reward/src/lib.rs

```diff
@@ -0,0 +1,295 @@
+
+pub trait Config: frame_system::Config + pallet_vesting::Config {
+    /// This module emits events, and hence, depends on the runtime's definition of event
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+
+    /// The balance type of the relay chain
+    type RelayChainBalance: Parameter + Member + AtLeast32BitUnsigned + Codec + Default + Copy +
+        MaybeSerializeDeserialize + Debug +
+        Into<BalanceOf<Self>>;
+
+    /// AccountId of the relay chain
+    type RelayChainAccountId: Parameter + Member + MaybeSerializeDeserialize + Debug + MaybeSerialize + Ord
+        + Default;
+
+    /// Admin or the module. I.e. this is necessary in cases, where the vesting parameters need
+    /// to be changed without an additional initialization.
+    type AdminOrigin: EnsureOrigin<Self::Origin>;
+
+    /// Weight information for extrinsics in this pallet
+    type WeightInfo: WeightInfo;
+}
+
+type BalanceOf<T> = <<T as pallet_vesting::Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+const MODULE_ID: ModuleId = ModuleId(*b"rwd/vest");
+
+decl_event!{
+    pub enum Event<T> where
+        AccountId = <T as frame_system::Config>::AccountId,
+        DirectReward = BalanceOf<T>,
+        VestedReward = BalanceOf<T>,
+        BlockNumber = <T as frame_system::Config>::BlockNumber,
+    {
+        /// Event emitted when a reward claim was processed successfully.
+        RewardClaimed(AccountId, DirectReward, VestedReward),
+        /// Event triggered when the reward module is ready to reward contributors
+        /// [vesting_start, vesting_period, conversion_rate, direct_payout_ratio]
+        RewardModuleInitialized(BlockNumber, BlockNumber, Perbill, Perbill),
+        /// Direct payout ratio for contributors has been updated
+        UpdateDirectPayoutRatio(Perbill),
+        /// Conversion rate from relay to native token has been updated
+        UpdateConversionRate(Perbill),
+        /// Vesting period has been updated
+        UpdateVestingPeriod(BlockNumber),
+        /// Start of vesting has been updated
+        UpadteVestingStart(BlockNumber),
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Config> as Reward {
+        /// The conversion rate between relay chain and native chain balances.
+        pub ConversionRate get(fn conversion_rate) build(|config: &GenesisConfig| Perbill::from_percent(config.conversion)): Perbill;
+        /// Which ratio of the rewards are payed directly. The rest is transferred via a vesting schedule.
+        pub DirectPayoutRatio get(fn direct_payout_ratio) build(|config: &GenesisConfig| Perbill::from_percent(config.direct_payout)): Perbill;
+        /// Over which period are the contributions vested.
+        pub VestingPeriod get(fn vesting_period) : T::BlockNumber;
+        /// At which block number does the vesting start .
+        pub VestingStart get(fn vesting_start) : T::BlockNumber;
+    } add_extra_genesis {
```

Comment:
do we need to derive from genisis here since they don't exist? Cant we have a default value with the possibility to change by the root extrinsic